plugins/modules/proxmox.py

```diff
@@ -727,10 +726,11 @@ def build_volume(
 
             # 1.3 If we have a host_path, we don't have storage, a volume, or a size
             vol_string = ",".join(
+                [vol_string] +
                 ([] if host_path is None else [host_path]) +
                 ([] if mountpoint is None else ["mp={0}".format(mountpoint)]) +
-                ([] if options is None else [map("=".join, options.items())]) +
-                ([] if not kwargs else [map("=".join, kwargs.items())])
+                ([] if options is None else list(map("=".join, options.items()))) +
+                ([] if not kwargs else list(map("=".join, kwargs.items())))
```

Comment:
Why use `map` at all, and not simply list comprehensions?
```suggestion
                ([] if options is None else ["{0}={1}".format(k, v) for k, v in options.items()]) +
                ([] if not kwargs else ["{0}={1}".format(k, v) for k, v in kwargs.items()])
```

Reply:
Since we now have to explicitly convert the map, using comprehensions makes more sense, I agree